module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,315 @@
+            try {
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                let getPollList = [
+                    pollsHelper.list
+                        (
+                            userId
+                        ),
+                    database.models.pollSubmissions.aggregate
+                        (
+                            [
+                                { "$match": { "userId": userId, isDeleted: false } },
+                                {
+                                    "$project": {
+                                        '_id': "$pollId",
+                                        'name': '$pollName'
+                                    }
+                                },
+                                { "$sort": { createdAt: 1 } }
+                            ])
+                ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+
+                        if (Array.isArray(response[0].data) && response[0].data.length > 0) {
+                            result = result.concat(response[0].data);
+                        }
+
+                        if (response[1].length > 0) {
+                            result = result.concat(response[1]);
+                        }
+                    });
+
+                let uniquePolls = [];
+
+                if (result.length > 0) {
+
+                    let mymap = new Map();
+                    uniquePolls = result.filter(el => {
+                        const val = mymap.get(el.name);
+                        if (val) {
+                            if (el._id < val) {
+                                mymap.delete(el.name);
+                                mymap.set(el.name, el._id);
+                                return true;
+                            } else {
+                                return false;
+                            }
+                        }
+                        mymap.set(el.name, el._id);
+                        return true;
+                    });
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: uniquePolls
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    /**
+    * Validate poll submission.
+    * @method
+    * @name validatePollSubmission
+    * @param {String} pollId - pollId
+    * @param {String} userId - userId
+    * @returns {String} - message.
+    */
+
+   static validatePollSubmission(pollId= "", userId= "") {
```

Comment:
@deepakr123 - Rename validatePollSubmission to a isPollSubmissionAllowed

Reply:
resolved